daemon/daemon.go

```diff
@@ -550,15 +550,17 @@ func (d *Daemon) compileBase() error {
 		return err
 	}
 
-	if option.Config.EnableIPv4 {
-		// Always remove masquerade rule and then re-add it if required
-		iptables.RemoveRules()
-		if option.Config.InstallIptRules {
-			if err := iptables.InstallRules(option.Config.HostDevice); err != nil {
-				return err
-			}
+	// Always remove masquerade rule and then re-add it if required
+	iptables.RemoveRules()
```

Comment:
Apologies, I haven't filtered through the commit messages, but this was previously only done if IPv4 is enabled. Is this deliberate?

Reply:
Yes. Previously all rules were IPv4 rules. Now some are IPv6 rules. Also, RemoveRules() can't fail, so no new error conditions are added. Besides, this is a little bit more correct now, as it is possible that node configuration has changed (the thing being processed here) so that IPv4 is changed from enabled to disabled. The old IPv4 rules should be removed also in that case.

`iptables.InstallRules()` now installs IPv4 rules only if IPv4 is enabled, and ditto for IPv6, so call to it should not be gated by IPv4 being enabled any more.